c++/src/kj/compat/gzip.c++

```diff
@@ -247,17 +247,13 @@ GzipAsyncOutputStream::GzipAsyncOutputStream(AsyncOutputStream& inner, int compr
 GzipAsyncOutputStream::GzipAsyncOutputStream(AsyncOutputStream& inner, decltype(DECOMPRESS))
     : inner(inner), ctx(kj::none) {}
 
-Promise<void> GzipAsyncOutputStream::write(const void* in, size_t size) {
-  ctx.setInput(in, size);
+Promise<void> GzipAsyncOutputStream::write(ArrayPtr<const byte> buffer) {
+  ctx.setInput(buffer.begin(), buffer.size());
   return pump(Z_NO_FLUSH);
 }
 
 Promise<void> GzipAsyncOutputStream::write(ArrayPtr<const ArrayPtr<const byte>> pieces) {
-  if (pieces.size() == 0) return kj::READY_NOW;
-  return write(pieces[0].begin(), pieces[0].size())
-      .then([this,pieces]() {
-    return write(pieces.slice(1, pieces.size()));
-  });
+  for (auto piece: pieces) co_await write(piece);
```

Comment:
as far as I remember this (and brotli.c++) was the only thing that I changed in this PR. The rest is straightforward interface adoption.